lib/v2/udn/breaking-news.js

```diff
@@ -0,0 +1,90 @@
+const got = require('@/utils/got');
+const { parseDate } = require('@/utils/parse-date');
+const timezone = require('@/utils/timezone');
+const cheerio = require('cheerio');
+const { URL } = require('url');
+
+module.exports = async (ctx) => {
+    const id = ctx.params.id;
+    const link = `/news/breaknews/1/${id}#breaknews`;
+    const name = await getLinkName(ctx, link);
+    const rootUrl = 'https://udn.com';
+    const response = await got(`${rootUrl}/api/more?page=1&channelId=1&cate_id=${id}&type=breaknews`);
+    const items = await Promise.all(
+        response.data.lists.map(async (item) => {
+            let link = item.titleLink.indexOf('http://') === 0 || item.titleLink.indexOf('https://') === 0 ? item.titleLink : `${rootUrl}${item.titleLink}`;
+            const linkUrl = new URL(link);
+            // cleanup query paramter
+            linkUrl.query = linkUrl.search = '';
+            link = linkUrl.toString();
+
+            const { author, description } = await ctx.cache.tryGet(link, async () => {
+                let result = await got(link);
+                // VIP article requires redirection
+                // e.g. https://udn.com/news/story/7331/6576320
+                const vip = result.data.match(/<script language=javascript>window\.location\.href="(https?:\/\/[^"]+")/);
+                if (vip !== null) {
+                    result = await got(vip[1]);
+                }
+                const $ = cheerio.load(result.data);
+                const figure = $('figure');
+                // e.g. https://udn.com/news/story/7001/6576479
+                const content = $('.article-content__editor');
+                // e.g. https://udn.com/news/story/7240/6576424
+                const body = $('.article-body__editor');
+
+                let author = '';
+                let description = '';
+                if (figure.length > 0) {
+                    description += figure.html();
+                }
+                if (content.length > 0) {
+                    author = $('.article-content__author a').text().trim();
+                    description += content.html();
+                } else if (body.length > 0) {
+                    author = $('.article-body__info span').text().trim();
+                    description += body.html();
+                }
+
+                return {
+                    author,
+                    description,
+                };
+            });
+
+            return {
+                title: item.title,
+                author,
+                description,
+                pubDate: timezone(parseDate(item.time.date, 'YYYY-MM-DD HH:mm'), +8),
+                link,
+            };
+        })
```

Comment:
The webpage has embedded JSON which you can make you of.
You can put other properties into cache as well.
```suggestion
        response.data.lists.map((item) => {
            let link = item.titleLink.indexOf('http://') === 0 || item.titleLink.indexOf('https://') === 0 ? item.titleLink : `${rootUrl}${item.titleLink}`;
            const linkUrl = new URL(link);
            // cleanup query paramter
            linkUrl.query = linkUrl.search = '';
            link = linkUrl.toString();

            return ctx.cache.tryGet(link, async () => {
                let result = await got(link);
                // VIP article requires redirection
                // e.g. https://udn.com/news/story/7331/6576320
                const vip = result.data.match(/<script language=javascript>window\.location\.href="(https?:\/\/[^"]+")/);
                if (vip) {
                    result = await got(vip[1]);
                }
                const $ = cheerio.load(result.data);
                const metadata = $('script[type="application/ld+json"]').eq(0).text().trim();
                const data = metadata.startsWith('[') ? JSON.parse(metadata)[0] : JSON.parse(metadata);

                // e.g. https://udn.com/news/story/7001/6576479
                const content = $('.article-content__editor');
                // e.g. https://udn.com/news/story/7240/6576424
                const body = $('.article-body__editor');

                let description = '';
                if (data.image) {
                    description += art(path.join(__dirname, 'templates/figure.art'), {
                        src: data.image.contentUrl,
                        alt: data.image.name,
                    });
                }
                if (content.length) {
                    description += content.html();
                } else if (body.length) {
                    description += body.html();
                }

                return {
                    title: item.title,
                    author: data.author.name,
                    description,
                    pubDate: timezone(parseDate(item.time.date, 'YYYY-MM-DD HH:mm'), +8),
                    category: [data.articleSelection, ...data.keywords.split(',')],
                    link,
                };
            });
        })
```
```html
<figure>
<picture><img src={{ src }} alt={{ alt }}></picture>
<figurecaption>{{ alt }}</figurecaption>
</figure>
```